create_dmd_release/create_dmd_release.d

```diff
@@ -511,7 +440,16 @@ void createRelease(string branch)
     if(exists(allExtrasDir)) copyDir(allExtrasDir, releaseDir);
     if(exists( osExtrasDir)) copyDir( osExtrasDir, releaseDir);
 
+    static void ensureIsClean(string repoDir)
+    {
+        const output = runCapture("cd "~quote(repoDir)~" && git status --porcelain");
+        if (output.length)
+            fail("Repo '"~repoDir~"' is dirty:\n" ~ output);
+    }
+
     // Copy sources
+    ensureIsClean(cloneDir~"/dmd");
+    ensureIsClean(cloneDir~"/phobos");
```

Comment:
... Here.

@schveiguy mentioned this last year, but also again recently in a DLF call. For some reason dmd releases have _always_ had `-dirty` in the version string. No surprises then that this fails.

https://pastebin.com/DfbFk8K4

The list appears to be arbitrary, so I suspect it's some automated unix2dos conversion (git itself is one known culprit for doing this).

Reply:
Running the commands individually inside the box, it differs at the first hurdle:
```
old mode 100755
new mode 100644
```
This is every file in the list, so it would appear `scp` command is at fault for not preserving permissions.

Reply:
https://github.com/dlang/installer/blob/master/create_dmd_release%2Fbuild_all.d#L153-L159

Instead invoking `scp -r - p` I get either of the following for various files.
```
scp: ... : set times: Input/output error
scp: ... : set times: Broken pipe
```

Reply:
https://github.com/dlang/installer/blob/master/create_dmd_release%2Fcreate_dmd_release.d#L297-L308

The best I can think of is to add
```
version (Windows)
    run("git config core.fileMode false");
```
After `git clean` for all repos in that function.

Reply:
FAOD, build succeeds with suggested filemode change.

Reply:
Is the version(Windows) necessary?

Reply:
If permissions of a file change when building on a proper filesystem, we want the build to fail.